packages/terra-form/src/Select.jsx

```diff
@@ -7,9 +7,18 @@ import './Select.scss';
 
 const propTypes = {
   /**
-   * List of choices to be selected.
+   * List of object key/value pairs for choices to be selected.
    */
-  choices: PropTypes.array.isRequired,
+  choices: PropTypes.arrayOf(PropTypes.shape({
+    value: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
+    ]).isRequired,
+    display: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
```

Comment:
I'm fine with oneOfType of string or number but I'll go with whatever the team consensus is.